🐛 Fix cron worker OOM-ing

azeemshaikh38 opened the pull request:
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Fixes OOM-ing issues of PubSub cron by writing Repo tarball to file instead of memory.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

oliverchang commented on clients/githubrepo/client.go:
```diff
@@ -47,27 +50,36 @@ func (client *Client) InitRepo(owner, repoName string) error {
 	defer resp.Body.Close()
 
-	respData, err := ioutil.ReadAll(resp.Body)
+	repoFile, err := os.OpenFile(repoFilename, os.O_CREATE|os.O_WRONLY, 0o644)
```
how about storing this on disk unpacked as well? the disk space savings probably aren't worth the repeated unpack/decompressions that the checks are doing.

azeemshaikh38 replied:
Yes, that is on my mind too. Just don't want to attempt it now since changing this would require me to update `CheckIfFileExists` and `CheckFileContent`.

Eventually, I plan to expose `ListFiles()` and `GetFileContent()` APIs instead of a `GetRepoArchive()` API. When I make this change I will update the underlying data to be stored uncompressed.

azeemshaikh38 commented:
Note to reviewers: I tested this locally and something seems off - the gzip/tar readers complain on reading the data from file. Will debug this further and update the PR.

azeemshaikh38 commented:
> Note to reviewers: I tested this locally and something seems off - the gzip/tar readers complain on reading the data from file. Will debug this further and update the PR.

Fixed. I was basically creating a single reader which got shared between threads. Updated it to read from same file but create a new reader per `GetRepoArchiveReader()` call.
